src/DataTable/DataTable.jsx

```diff
@@ -14,38 +14,46 @@ export class DataTable extends Component {
   }
   
   static defaultProps = {
-    autoResize: false
+    autoResize: true
   }
-  
-  handleResize (e) {
+
+  handleResize (grid) {
     if (this.autoResize) {
-      this.refs.grid.api.sizeColumnsToFit()
+      grid.api.sizeColumnsToFit()
     }
   }
-
+  
   componentDidMount () {
-    window.addEventListener('resize', this.handleResize.bind(this))
+    const grid = this.refs.grid
+  
+    if (this.autoResize) {
+      grid.api.sizeColumnsToFit()
+    }
+  
+    this.boundResize = this.handleResize.bind(this, grid)
+  
+    window.addEventListener('resize', this.boundResize)
   }
-
+  
   componentWillUnmount () {
-    window.removeEventListener('resize', this.handleResize.bind(this))
+    window.removeEventListener('resize', this.boundResize)
   }
-
+  
   headerCellRendererFunc (params) {
     const api = params.api
     const cb = document.createElement('input')
     cb.setAttribute('type', 'checkbox')
     cb.setAttribute('class', 'ag-selection-checkbox')
-
+  
     const label = document.createElement('label')
     const title = document.createTextNode(params.colDef.headerName)
     const span = document.createElement('span')
     span.setAttribute('class', 'ag-cell-value')
-    
+  
     label.appendChild(cb)
     label.appendChild(span)
     label.appendChild(title)
-    
+  
     cb.addEventListener('change', function (e) {
       if (this.checked) {
         api.selectAll()
```

Comment:
On line 79 and 81,  you have this:

```javascript
const {autoResize, checkboxColumn, columns, data, stripeRows} = this.props

this.autoResize = autoResize

// that can change to this, given the above changes (remove autoResize from destructure, dont set this.autoResize)

const {checkboxColumn, columns, data, sriptRows} = this.props
```

on line 83, you have:

```javascript
let newColumns = columns

// you can just leave it undefined since `Array.map` returns a new array

let newColumns
```

on line 100 you have:

```javascript
if (stripeRows === true) {
  gridProps.getRowClass = (params) => {
    if (params.rowIndex % 2 === 1) {
      return 'ag-row-colored'
    }
  }
}

// this can be

if (stripeRows) {
  gridProps.getRowClass = (params) => (
    params.rowIndex % 2 === 1 ? 'ag-row-colored' : '' // <-- if necessary this could also be null or undefined instead of ''
  )
}
```

on line 109 you have

```javascript
if (checkboxColumn != null && checkboxColumn !== false) {
  newColumns = [Object.assign({
    checkboxSelection: true,
    headerCellRenderer: this.headerCellRendererFunc,
    headerName: '',
    suppressMovable: true,
    suppressResize: true,
    suppressSorting: true,
    width: 60
  }, checkboxColumn), ...newColumns]
       
  // Setup the grid so that it properly checks/unchecks
  Object.assign(gridProps, {
    onRowClicked: (event) => {
      const rowNode = event.node
      rowNode.setSelected(!rowNode.isSelected())
    },
    rowSelection: 'multiple',
    suppressRowClickSelection: true
  })
}

// that can be

if (checkboxColumn) {
  newColumns = [{
    checkboxSelection: true,
    headerCellRenderer: this.headerCellRendererFunc,
    headerName: '',
    suppressMovable: true,
    suppressResize: true,
    suppressSorting: true,
    width: 60
  }, ...newColumns]
  
  gridProps = {
    ...gridProps,
    onRowClicked: (event) => {
      const rowNode = event.node
      rowNode.setSelected(!rowNode.isSelected())
    },
    rowSelection: 'multiple',
    suppressRowClickSelection: true
  }
}
```